### PR TITLE
refactor api code

### PR DESCRIPTION
**Pull request by zaziemo**
As a necessity for implementing elasticsearch with my favourite gem we have to refactor the api code. Until now the api was created by overwriting the as_json method, with this PR it is covered by a serializer.

### CODE REVIEW

**Comment by zaziemo** [comment]
@tyranja could you please double check and merge?

File: spec/controllers/api/v1/profiles_controller_spec.rb [hunk 1]
```diff
@@ -14,34 +15,119 @@
...
+    end
+
+    it 'should contain attribute updated_at' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['updated_at']).to eq(profile1.updated_at.as_json)
+    end
+
+    it 'should contain attribute medialinks' do
+      get :index, ids: [profile1.id], format: :json
+      expect(data[0]['medialinks']).to eq([])
```
Comment (tyranja) [inline spec/controllers/api/v1/profiles_controller_spec.rb]: One day we have to add medialinks etc there to check them as well. but for now I guess its ok.

File: app/models/profile.rb [hunk 2]
```diff
@@ -120,26 +121,6 @@ def update_with_password(params, *options)
...
-      'website'
-    ).merge(
-      'medialinks' => medialinks,
-      'topics' => topics.map(&:name),
-      'picture' => picture,
-      'bio' => bio_translations,
-      'main_topic' => main_topic_translations
-    )
-  end
-
```
Comment (tyranja) [inline app/models/profile.rb]: The medialink model has an as_json method as well. Do we still need that?

**Comment by zaziemo** [comment]
@tyranja you are right with your v2/v1 comment. But since no one was using v1 I thought it would be ok :)

**Pull request merged**